Common/Types/include/Types/ValuePack.h

```diff
@@ -0,0 +1,170 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef TYPES_VALUEPACK_H
+#define TYPES_VALUEPACK_H
+/// @file   ValuePack.h
+/// @author Matthias Richter, Sandro Wenzel
+/// @since  2017-09-28
+/// @brief  A compact representation of bits from multiple values
+
+#include <boost/mpl/vector_c.hpp>
+#include <boost/mpl/vector.hpp>
+#include "boost/mpl/size.hpp"
+#include <boost/mpl/pair.hpp>
+#include <boost/mpl/push_back.hpp>
+#include <boost/mpl/int.hpp>
+#include "boost/mpl/at.hpp"
+#include <boost/mpl/fold.hpp>
+#include <boost/mpl/placeholders.hpp>
+
+namespace mpl = boost::mpl;
+using _1 = mpl::_1;
+using _2 = mpl::_2;
+
+namespace o2 {
+namespace types {
+
+//////////////////////////////////////////////////////////////////////////
+/// Recurssively build the code
+template<typename Rep, typename Iterator, typename End>
+struct CodeInit {
+  using Shift = typename boost::mpl::deref<Iterator>::type;
+  static const Rep mask = (Rep(0x1) << Shift::value) - 1;
+
+  template<typename... Types>
+  static Rep value(size_t arg, Types... args) {
+    return (arg & mask) | CodeInit<Rep, typename boost::mpl::next<Iterator>::type,
+                                   End
+                                   >::value(args...) << Shift::value;
+  }
+};
+
+// specialization terminating the recursive meta function
+template<typename Rep, typename End>
+struct CodeInit<Rep, End, End> {
+  static Rep value() {return 0;}
+};
+
+/**
+* @class ValuePack
```

Comment:
I would add a somewhat longer comment explaining this class:
```
ValuePack is a generic template class packing/unpacking multiple integral values of given 
bit sizes into/from an underlying storage type Rep.
```

Reply:
good, will add this, anything else we should include in the first version?